helm/flux-app/templates/_helpers.tpl

```diff
@@ -55,3 +55,23 @@ app.kubernetes.io/instance: {{ .Release.Name | quote }}
 {{- define "controllerVolumeName" -}}
 {{- printf "%s-controller-%s-volume" (include "name" .) .volumeName -}}
 {{- end -}}
+
+{{- define "appPlatform.twoStepInstall" -}}
+{{- $is_chart_operator := lookup "application.giantswarm.io/v1alpha1" "Chart" "giantswarm" "chart-operator" -}}
+{{- $is_chart_operator_bad := true }}
+{{- if $is_chart_operator }}
+{{- $is_chart_operator_bad = (semverCompare "< 2.26.0-0" $is_chart_operator.spec.version) }}
+{{- end }}
+
+{{- $is_this_chart_cr := lookup "application.giantswarm.io/v1alpha1" "Chart" "giantswarm" . -}}
+{{- $is_outside_app_platform := true }}
+{{- if $is_this_chart_cr }}
+{{- $is_outside_app_platform = false }}
+{{- end }}
+
+{{- if or $is_chart_operator_bad $is_outside_app_platform }}
```

Comment:
If any:
- don't find the Chart Operator Chart CR,
- find the Chart Operator Chart CR, but it runs old version that does not support two-step install,
- don't find this app Chart CR
then render two-step install as unsupported, as it means the app is most probably installed outside the App Platform.